scanner/services/matcher.go

```diff
@@ -164,3 +170,44 @@ func (s *matcherService) notes(ctx context.Context, vr *v4.VulnerabilityReport)
 
 	return nil
 }
+
+func (s *matcherService) GetSBOM(ctx context.Context, req *v4.GetSBOMRequest) (*v4.GetSBOMResponse, error) {
+	ctx = zlog.ContextWithValues(ctx, "component", "scanner/service/matcher.GetSBOM")
+
+	if err := validators.ValidateGetSBOMRequest(req); err != nil {
+		return nil, errox.InvalidArgs.CausedBy(err)
+	}
+
+	// We only support container image resources for now.
+	digestStr, found := strings.CutPrefix(req.GetHashId(), "/v4/containerimage/")
+	if !found {
+		return nil, errox.InvalidArgs.CausedByf("invalid hash id %q, only container images are supported", req.GetHashId())
+	}
+
+	ccDigest, err := claircore.ParseDigest(digestStr)
+	if err != nil {
+		return nil, errox.InvalidArgs.CausedByf("invalid digest in hash id %q: %v", req.GetHashId(), err)
+	}
+
+	zlog.Info(ctx).Msgf("generating SBOM for index report %q with contents (%d dists, %d envs, %d pkgs, %d repos)",
+		req.GetHashId(),
+		len(req.GetContents().GetDistributions()),
+		len(req.GetContents().GetEnvironments()),
+		len(req.GetContents().GetPackages()),
+		len(req.GetContents().GetRepositories()),
+	)
+
+	ir, err := s.parseIndexReport(req.GetContents())
```

Comment:
so we don't utilize the remote indexer? can you add a comment to make it clear that we do not support using the remote indexer for this and why we don't?